<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageProcessor</class>
 <widget class="QMainWindow" name="ImageProcessor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1324</width>
    <height>760</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1324</width>
    <height>522</height>
   </size>
  </property>
  <property name="sizeIncrement">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Digital Image Processing App</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>1324</width>
     <height>512</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="sizeIncrement">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="mainLabel">
      <property name="text">
       <string>Choose image from file:</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QScrollArea" name="imageScrollArea">
        <property name="minimumSize">
         <size>
          <width>1024</width>
          <height>512</height>
         </size>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QAbstractScrollArea::AdjustToContents</enum>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="imageScrollAreaContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>1040</width>
           <height>640</height>
          </rect>
         </property>
         <widget class="QLabel" name="imageLabel">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>1306</width>
            <height>512</height>
           </rect>
          </property>
          <property name="minimumSize">
           <size>
            <width>1024</width>
            <height>512</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="processVertLayout">
        <item>
         <widget class="QScrollArea" name="processScrollArea">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QAbstractScrollArea::AdjustToContents</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="processScrollAreaContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>267</width>
             <height>619</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>267</width>
             <height>510</height>
            </size>
           </property>
           <property name="layoutDirection">
            <enum>Qt::RightToLeft</enum>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>231</width>
              <height>626</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="vertLayout">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="sizeConstraint">
              <enum>QLayout::SetMinimumSize</enum>
             </property>
             <item>
              <widget class="QGroupBox" name="resizeGroup">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>145</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>145</height>
                </size>
               </property>
               <property name="title">
                <string>Resize</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="flat">
                <bool>false</bool>
               </property>
               <property name="checkable">
                <bool>false</bool>
               </property>
               <widget class="QLabel" name="resizeLabel">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>30</y>
                  <width>101</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Resize Factor:</string>
                </property>
               </widget>
               <widget class="QSpinBox" name="resizeFactor">
                <property name="geometry">
                 <rect>
                  <x>150</x>
                  <y>30</y>
                  <width>51</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="alignment">
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
                <property name="specialValueText">
                 <string/>
                </property>
                <property name="suffix">
                 <string/>
                </property>
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>900</number>
                </property>
                <property name="singleStep">
                 <number>1</number>
                </property>
                <property name="value">
                 <number>100</number>
                </property>
               </widget>
               <widget class="QLabel" name="upsampleLabel">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>60</y>
                  <width>161</width>
                  <height>17</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Interpolation Method:</string>
                </property>
               </widget>
               <widget class="QComboBox" name="upsampleCombo">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="geometry">
                 <rect>
                  <x>70</x>
                  <y>80</y>
                  <width>151</width>
                  <height>25</height>
                 </rect>
                </property>
                <property name="insertPolicy">
                 <enum>QComboBox::InsertAtBottom</enum>
                </property>
                <property name="frame">
                 <bool>true</bool>
                </property>
                <item>
                 <property name="text">
                  <string>Nearest Neighbor</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Linear X-Direction</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Linear Y-Direction</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Bilinear</string>
                 </property>
                </item>
               </widget>
               <widget class="QPushButton" name="applyResize">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>110</y>
                  <width>211</width>
                  <height>25</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
               <widget class="QLabel" name="label">
                <property name="geometry">
                 <rect>
                  <x>210</x>
                  <y>30</y>
                  <width>16</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="text">
                 <string>%</string>
                </property>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="spacialFilterGroup">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>180</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>180</height>
                </size>
               </property>
               <property name="title">
                <string>Spacial Filters</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <widget class="QComboBox" name="spacialFilterComboBox">
                <property name="geometry">
                 <rect>
                  <x>60</x>
                  <y>30</y>
                  <width>161</width>
                  <height>25</height>
                 </rect>
                </property>
                <item>
                 <property name="text">
                  <string>Smoothing</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Sharpening Laplacian</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>High-Boost</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Histogram Equalization</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Median</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Max</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Min</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Arithmetic Mean</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Geometric Mean</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Harmonic Mean</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Contraharmonic Mean</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Midpoint</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Alpha-Trimmed Mean</string>
                 </property>
                </item>
               </widget>
               <widget class="QLabel" name="filterLabel">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>30</y>
                  <width>67</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Filter:</string>
                </property>
               </widget>
               <widget class="QLabel" name="spacialSizeLabel">
                <property name="geometry">
                 <rect>
                  <x>38</x>
                  <y>60</y>
                  <width>101</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Kernel Size: 3 x</string>
                </property>
               </widget>
               <widget class="QSpinBox" name="spacialSizeSpin">
                <property name="geometry">
                 <rect>
                  <x>142</x>
                  <y>60</y>
                  <width>48</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="alignment">
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
                <property name="minimum">
                 <number>3</number>
                </property>
                <property name="maximum">
                 <number>9</number>
                </property>
                <property name="singleStep">
                 <number>2</number>
                </property>
               </widget>
               <widget class="QLabel" name="hbConstLabel">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>90</y>
                  <width>151</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="text">
                 <string>HB + CH Mean Const:</string>
                </property>
               </widget>
               <widget class="QDoubleSpinBox" name="hbConstSpin">
                <property name="geometry">
                 <rect>
                  <x>160</x>
                  <y>90</y>
                  <width>61</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="alignment">
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
                <property name="decimals">
                 <number>1</number>
                </property>
                <property name="minimum">
                 <double>-10.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="value">
                 <double>1.000000000000000</double>
                </property>
               </widget>
               <widget class="QPushButton" name="applySpacialFilter">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>150</y>
                  <width>211</width>
                  <height>25</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
               <widget class="QSpinBox" name="atSpinBox">
                <property name="geometry">
                 <rect>
                  <x>167</x>
                  <y>120</y>
                  <width>51</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="singleStep">
                 <number>2</number>
                </property>
               </widget>
               <widget class="QLabel" name="atConstLabel">
                <property name="geometry">
                 <rect>
                  <x>6</x>
                  <y>120</y>
                  <width>151</width>
                  <height>20</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Alpha-Trim Constant:</string>
                </property>
               </widget>
               <zorder>spacialFilterComboBox</zorder>
               <zorder>filterLabel</zorder>
               <zorder>spacialSizeSpin</zorder>
               <zorder>hbConstLabel</zorder>
               <zorder>hbConstSpin</zorder>
               <zorder>applySpacialFilter</zorder>
               <zorder>spacialSizeLabel</zorder>
               <zorder>atSpinBox</zorder>
               <zorder>atConstLabel</zorder>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="bitPlaneGroup">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>124</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>124</height>
                </size>
               </property>
               <property name="title">
                <string>Alter Bit Plane</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <widget class="QLabel" name="bitPlaneLabel">
                <property name="geometry">
                 <rect>
                  <x>20</x>
                  <y>30</y>
                  <width>71</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Bit Plane:</string>
                </property>
               </widget>
               <widget class="QSpinBox" name="bitPlane">
                <property name="geometry">
                 <rect>
                  <x>100</x>
                  <y>30</y>
                  <width>48</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="maximum">
                 <number>7</number>
                </property>
               </widget>
               <widget class="QPushButton" name="applyBitPlane">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>90</y>
                  <width>211</width>
                  <height>25</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
               <widget class="QCheckBox" name="bitPlaneZero">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="geometry">
                 <rect>
                  <x>110</x>
                  <y>60</y>
                  <width>92</width>
                  <height>23</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Zero out?</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="colorResolution">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>95</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>95</height>
                </size>
               </property>
               <property name="title">
                <string>Color Binning</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <widget class="QLabel" name="bitnessLabel">
                <property name="geometry">
                 <rect>
                  <x>40</x>
                  <y>30</y>
                  <width>61</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Bitness:</string>
                </property>
               </widget>
               <widget class="QSpinBox" name="bitness">
                <property name="geometry">
                 <rect>
                  <x>140</x>
                  <y>30</y>
                  <width>48</width>
                  <height>26</height>
                 </rect>
                </property>
                <property name="alignment">
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>8</number>
                </property>
                <property name="value">
                 <number>8</number>
                </property>
               </widget>
               <widget class="QPushButton" name="applyColorBin">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>60</y>
                  <width>211</width>
                  <height>25</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="histEqualButton">
               <property name="text">
                <string>Histogram Equalization</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="convertToGray">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Convert to Grayscale</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="saveProcImage">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Save New Image</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="useProcImage">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Use New Image</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1324</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_Image"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionOpen_Image">
   <property name="text">
    <string>Open Image...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
